packages/message-relayer/src/relay-tx.ts

```diff
@@ -421,3 +421,111 @@ export const getMessagesAndProofsForL2Transaction = async (
 
   return messagePairs
 }
+
+/**
+ * Allows for proof generation of pre-regenesis L2->L1 messages, by retrieving proofs from
+ * The genesis state (block 0) of the post-regenesis chain. This is required because the
+ * history is wiped during regnesis, so old inclusion proofs would no longer work.
+ *
+ * @param l1RpcProvider L1 RPC provider.
+ * @param l2RpcProvider L2 RPC provider of the POST-REGENESIS chain.
+ * @param legacyL2Transaction A PRE-REGENESIS L2 transaction which sent some L2->L1 messages.
+ * @param legacyMessages The L2->L1 messages which were sent by the legacy L2 transaction.
+ * @param l1StateCommitmentChainAddress Address of the POST-REGENESIS StateCommitmentChain.
+ * @param l2CrossDomainMessengerAddress Address of the L2CrossDomainMessenger.
+ * @returns An array of messages sent in the transaction and a proof of inclusion for each.
+ */
+export const getLegacyProofsForL2Transaction = async (
+  l1RpcProvider: ethers.providers.JsonRpcProvider | string,
+  l2RpcProvider: ethers.providers.JsonRpcProvider | string,
+  legacyL2Transaction: Transaction,
```

Comment:
IMO it would be ideal if we used the same function interface as the `getMessagesAndProofsForL2Transaction` function.

Reply:
This is fundamentally not possible, since it either requires both L2 providers, or providing a cached input from the historic provider (as is done here).

I originally wrote it as the same but just adding the L2 provider as a new argument.  However, @K-Ho @SmitV and I decided to change it because:
1. We do not want a dependency on the historic L2 provider being up in perpetuity.
2. We are caching this up front so that we can index by user.
3. We can't cache the proof data before regenesis starts, since it requires the new provider and genesis state root.

Reply:
Ok that's fine, if @SmitV has tested this function and it works then I will approve

Reply:
Yep tested and it works